ubuntu_image/parser.py

```diff
@@ -347,9 +347,15 @@ def parse(stream_or_string):
         bootloader_seen |= (bootloader is not None)
         image_id = image_spec.get('id')
         structures = []
+        structure_names = []
```

Comment:
This is more of a nitpick (/preference) really, but could we use a set() instead of a list here? Since the only thing we do is a check  `name in structure_names` and append, we could use a set() + `structure_names.add(name)` instead.

Reply:
Thanks for the suggestion. Updated & pushed.